pkg/paioavail/paio_avail_listener.go

```diff
@@ -266,35 +278,48 @@ func (a AvailListener) TableTennis(ctx context.Context,
 			return nil, err
 		}
 		for i := range inputs {
-			inputs[i].Index = i + int(inputCount)
-			if inputs[i].AppContract != a.ApplicationAddress {
+			inputExtra := inputs[i]
+			inputExtra.Input.Index = uint64(i) + inputCount
+			var app *model.Application = nil
+			for _, appr := range apps {
+				if inputExtra.AppContract.Hex() == appr.IApplicationAddress.Hex() {
+					app = &appr
+					break
+				}
+			}
+			if app == nil {
 				slog.Warn("Skipping input",
-					"appContract", inputs[i].AppContract.Hex(),
-					"expected", a.ApplicationAddress.Hex(),
-					"msgSender", inputs[i].MsgSender.Hex(),
-					"payload", inputs[i].Payload,
+					"appContract", inputExtra.AppContract.Hex(),
+					"expected", apps,
 				)
 				continue
 			}
 			// The chainId information does not come in Paio's batch.
-			_, err = a.InputRepository.Create(ctx, inputs[i])
+			input := inputExtra.Input
+			input.EpochApplicationID = app.ID
+
+			epoch, err := a.EpochRepository.GetLatestOpenEpoch(ctx)
```

Comment:
The InputService (within the base reader) should take care of handling the epoch

Reply:
and by the way, we need to specify which app we are reading the current epoch

Reply:
Service: https://github.com/Calindra/rollups-base-reader/pull/6